MT-CTF/capturetheflag
# Display actual kill weapons in kills list

![screenshot_20190712_104651](https://user-images.githubusercontent.com/36130650/61104051-6a78e880-a492-11e9-9f86-6bd1b15dc4e1.png)

### Use-cases

- Bragging rights: Imagine how awesome it'd be to kill an enemy with a sapling and have it show up in the kills list. :)

(Take the above screenshot for example - I pwned ClobberXD using furniture and bathroom supplies)

### Implementation

#### Checking for grenade kill

- If stack name is empty:
  - If `damage_groups.grenade` then kill weapon = `shooter:grenade`
  - Else, kill weapon = `""` (i.e. bare fists)

#### Retrieving image of kill weapon

- If kill weapon = `""`, image = `wieldhand.png`
- Else, select image from item def with the following order of preference:
  - If `inventory_image` and `inventory_image ~= ""`, image = `inventory_image`
  - Elseif `tiles[1]`
    - If `tiles[1]` is a table, image = `tiles[1].name`
    - Else, image = `tiles[1]`
  - Else image = `ctf_events_fallback.png` (fallback texture)

### To test

- Open up two instances of MT.
- Kill the other player with a variety of items (craftitems, tools, nodes, nodeboxes, etc.), _and_ with bare fists.
- Check the kills list to see if all the kill weapons are displayed correctly.
